### PR TITLE
Spelling ql

### PR DESCRIPTION
**Pull request by jsoref**
follow-up to https://github.com/github/codeql/pull/10743

### CODE REVIEW

File: ql/ql/src/codeql_ql/ast/internal/AstNodes.qll [hunk 1]
```diff
@@ -60,7 +60,7 @@ newtype TAstNode =
   TPredicateExpr(QL::PredicateExpr pe) or
   TAnnotation(QL::Annotation annot) or
   TAnnotationArg(QL::AnnotArg arg) or
-  TYamlCommemt(Yaml::Comment yc) or
+  TYamlComment(Yaml::Comment yc) or
```
Comment (jsoref) [inline ql/ql/src/codeql_ql/ast/internal/AstNodes.qll:63]: Hopefully not a problem

File: ql/ql/src/codeql_ql/ast/internal/Builtins.qll [hunk 2]
```diff
@@ -61,12 +61,12 @@ predicate isBuiltinMember(string qual, string ret, string name, string args) {
 }
 
 module BuiltinsConsistency {
-  query predicate noBuildinParse(string sig) {
+  query predicate noBuiltinParse(string sig) {
```
Comment (jsoref) [inline ql/ql/src/codeql_ql/ast/internal/Builtins.qll:64]: and similarly

File: ql/ql/src/queries/style/RegexpInsteadOfPattern.ql [hunk 3]
```diff
@@ -3,7 +3,7 @@
  * @description The `matches` builtin predicate takes a special pattern format as an input, not a regular expression.
  * @kind problem
  * @problem.severity warning
- * @id ql/rexexp-pattern
+ * @id ql/regex-pattern
```
Comment (jsoref) [inline ql/ql/src/queries/style/RegexpInsteadOfPattern.ql:6]: ?

File: ql/ql/src/codeql_ql/ast/internal/AstNodes.qll (hunk 1 above)
Comment (erik-krogh) [reply ql/ql/src/codeql_ql/ast/internal/AstNodes.qll:63]: Not a problem.  
We don't need to follow any kind of deprecation policy inside the `ql/` folder, as it's not part of any product.  

We should be good as long as the CI passes.

File: ql/ql/src/codeql_ql/style/MisspellingQuery.qll [hunk 4]
```diff
@@ -11,7 +11,7 @@ predicate misspelling(string wrong, string right, string mistake) {
 }
 
 predicate additional_typos(string wrong, string right) {
-  wrong = "tranformer" and right = "transformer"
+  wrong = "transformer" and right = "transformer"
```
Comment (erik-krogh) [inline ql/ql/src/codeql_ql/style/MisspellingQuery.qll:14]:
```suggestion
  wrong = "tranformer" and right = "transformer"
```

This was an intentional typo.

File: ql/ql/src/queries/style/RegexpInsteadOfPattern.ql (hunk 3 above)
Comment (erik-krogh) [reply ql/ql/src/queries/style/RegexpInsteadOfPattern.ql:6]: Both are fine.

File: ql/ql/src/codeql_ql/style/MisspellingQuery.qll (hunk 4 above)
Comment (jsoref) [reply ql/ql/src/codeql_ql/style/MisspellingQuery.qll:14]: Sigh. I knew to look out for that file but missed reviewing it.

**Review by erik-krogh — APPROVED** [review]

**Pull request merged**